app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,74 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show
```

Comment:
Seems like when you load data from this action, there is a N+1 query. Please double-check `Post.find(params[:id])`, probably for this action you should use preloading of associations(e.g. ActiveRecord `includes` method)

Reply:
Please install gem 'bullet', it will help you to find N+1 query problem. This is problem is important.